src/python/grpcio_health_checking/grpc_health/v1/health.py

```diff
@@ -27,7 +28,7 @@ class _Watcher():
 
     def __init__(self):
         self._condition = threading.Condition()
-        self._responses = list()
+        self._responses = collections.deque()
```

Comment:
So, after suggesting `deque` here, I discovered that [`Queue` is basically just a `deque` and some locking machinery.](https://github.com/python/cpython/blob/master/Lib/queue.py#L205) It's also recommended as [the default inter-thread communication mechanism](https://github.com/google/styleguide/blob/gh-pages/pyguide.md#218-threading) by the style guide. Not wholly necessary, but a thought.